x/perpetual/keeper/settle_funding_fee_distribution.go

```diff
@@ -48,48 +42,53 @@ func (k Keeper) SettleFundingFeeDistribution(ctx sdk.Context, mtp *types.MTP, po
 	if mtp.Position == types.Position_LONG {
 		// Ensure liabilitiesLong is not zero to avoid division by zero
 		if liabilitiesLong.IsZero() {
-			return sdk.Coin{}, types.ErrAmountTooLow
+			return types.ErrAmountTooLow
 		}
 		fundingFeeShare = sdk.NewDecFromInt(mtp.Liabilities).Quo(sdk.NewDecFromInt(liabilitiesLong))
 		totalFund = short
 	} else {
 		// Ensure liabilitiesShort is not zero to avoid division by zero
 		if liabilitiesShort.IsZero() {
-			return sdk.Coin{}, types.ErrAmountTooLow
+			return types.ErrAmountTooLow
 		}
 		fundingFeeShare = sdk.NewDecFromInt(mtp.Liabilities).Quo(sdk.NewDecFromInt(liabilitiesShort))
 		totalFund = long
 	}
 
 	// if funding fee share is zero, skip mtp
 	if fundingFeeShare.IsZero() {
-		return sdk.Coin{}, nil
+		return nil
 	}
 
 	// calculate funding fee amount
 	fundingFeeAmount := sdk.NewCoin(baseCurrency, totalFund.Mul(fundingFeeShare).TruncateInt())
-	toPay := sdk.Coin{}
 
-	if balance.Amount.LT(fundingFeeAmount.Amount) {
-		toPay = fundingFeeAmount
-	} else {
-		// transfer funding fee amount to mtp address
-		if err := k.bankKeeper.SendCoins(ctx, fundingFeeCollectionAddress, mtpAddress, sdk.NewCoins(fundingFeeAmount)); err != nil {
-			return sdk.Coin{}, err
-		}
+	// update mtp custody
+	mtp.Custody = mtp.Custody.Add(fundingFeeAmount.Amount)
+
+	// update pool custody balance
+	err = pool.UpdateCustody(ctx, mtp.CustodyAsset, fundingFeeAmount.Amount, false, mtp.Position)
```

Comment:
@amityadav0 I believe the boolean should be `true` here for incrementing the custody value rather than `false`, correct me if I am wrong, thanks

Reply:
we are sending fees from pool to mtp, so it should be reduced, right ? In fee collection we should increase it as we are collecting

Reply:
@amityadav0 nope the fees are collected from mtps to the funding address see here https://github.com/elys-network/elys/pull/844/files#diff-943de122eee4295bcf39ae75c877dc118541e577b8cd1ecb4b64ee2cfae9d8e6R62

here we are dealing with the distribution process, so if any given mtp earns with the fees then the fees should be added to its mtp `custody`

Reply:
ah, got it

Reply:
updated